tests/duplicate_event/run.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+set -eu
+
+cur=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+source $cur/../_utils/test_prepare
+WORK_DIR=$TEST_DIR/$TEST_NAME
+TASK_NAME="test"
+SQL_RESULT_FILE="$TEST_DIR/sql_res.$TEST_NAME.txt"
+
+API_VERSION="v1alpha1"
+
+function run() {
+	# 1. test sync fetch binlog met error and reset binlog streamer with remote binlog
+
+	# with a 5 rows insert txn: 1 * GTID + 1 * BEGIN + 5 * (Table_map + Write_rows) + 1 * XID
+	# here we fail at the third write rows event, sync should retry and auto recover without any duplicate event
+	export GO_FAILPOINTS="github.com/pingcap/dm/syncer/GetEventErrorInTxn=13*return(3)"
```

Comment:
there should be format_description event and previoud_gtid event so it's 15, so maybe we should check `failed to get event` in log.
But seems 13 also triggered "discard event already consumed" 🤔

Reply:
The failpoint will trigger the error at the third WriteRowsEvent, so 13 is a upper bound the ensure that it won't impact the next txn

Reply:
1. format_description
2. previous_gtid
3. gtid
4. query (begin)
5 ~ 8
9. table_map 3
10. write_rows 3
11 ~ 14
15. xid

now I see 13 times is enough

Reply:
https://github.com/pingcap/dm/pull/2047/files#r699233692 I think in this comment, @lance6716 also prefer to just skip the duplicate event instead of relay on safe-mode.